Automattic/jetpack
# AI Featured Image: enable Stable Diffusion to all users and improve tracking information

~Depends on D149604-code.~ The required code got merged already.

## Proposed changes:

* Enable stable diffusion to 100% of the site base
* Add site type to the events being tracked
* Add generation model to the events being tracked
* Drop the need of the `jetpack_experimental_image_generation_enabled ` filter to enable the Stable Diffusion generation
* Change the endpoint used to generate the images, to use the same that provides Dall-e-3 generation
* Refactor code now that both endpoints are the same

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



Yes, it's including the site type and the image generation model on the events tracked by the Featured Image Generation tool.

## Testing instructions:






* ~Sandbox the `public-api` and apply the D149604-code diff to your sandbox~
* Test it on a Simple site, a Jetpack site and, if possible, an Atomic site too
* Run this branch on your testing site
   * For Atomic sites, use the [Jetpack Beta Tester plugin](https://github.com/Automattic/jetpack-beta) to run the branch, the same way you would do on a JN site
* Make sure beta blocks are disabled and you are running production blocks on your testing site
   * For Simple sites, use it unproxied so you get the production blocks
* Go to the block editor, open the browser console and enable debug messages with `localStorage.setItem( 'debug', '*' );`
* Add some content to the editor, than look for the AI Featured Image tool and click to generate the image
* Confirm an image is generated
* Confirm the generation event was triggered containing the site type and the image generation model
   * the site type needs to be the type of the site you are using for tests
   * the generation model needs to be `stable-diffusion` since you are using the beta variation
* Confirm you can use the tool as expected, with follow ups and every other buttons working as expected
